src/html5/parser.rs

```diff
@@ -630,23 +635,24 @@ impl<'chars> Html5Parser<'chars> {
     fn process_html_content(&mut self) {
         if self.ignore_lf {
             if let Token::Text(value) = &self.current_token {
-                if value.eq(&"\n".to_string()) {
-                    self.current_token = self.fetch_next_token();
+                if value.starts_with('\n') {
```

Comment:
instead of fetching the next token, we check if the current token has a \n. If so, we modify the current token to remove the \n and continue as usual.